### PR TITLE
fix: confirm when deleting template

### LINKED ISSUE #2830: `coder templates delete` does not confirm
It just deletes the files!  That's dangerous!

### PR DESCRIPTION
**Pull request by ketang**
fixes #2830

### CODE REVIEW

File: cli/templatedelete.go [hunk 1]
```diff
@@ -57,17 +67,19 @@ func templateDelete() *cobra.Command {
...
-				if err != nil {
-					return xerrors.Errorf("get template by name: %w", err)
-				}
-
-				templates = append(templates, template)
+			// Confirm deletion of the template.
+			_, err = cliui.Prompt(cmd, cliui.PromptOptions{
+				Text:      fmt.Sprintf("Delete these templates: %s?", cliui.Styles.Code.Render(strings.Join(templateNames, ", "))),
+				IsConfirm: true,
+				Default:   "no",
```
Comment (kylecarbs) [inline cli/templatedelete.go:79]: All of our other confirmation prompts default to "yes". We should probably make workspaces default to "no" as well.

File: cli/templatedelete.go [hunk 2]
```diff
@@ -57,17 +67,19 @@ func templateDelete() *cobra.Command {
...
 
-			for _, templateName := range templateNames {
-				template, err := client.TemplateByName(ctx, organization.ID, templateName)
-				if err != nil {
-					return xerrors.Errorf("get template by name: %w", err)
-				}
-
-				templates = append(templates, template)
+			// Confirm deletion of the template.
+			_, err = cliui.Prompt(cmd, cliui.PromptOptions{
```
Comment (kylecarbs) [inline cli/templatedelete.go:76]: I assume a test will have to be changed here too.

File: cli/templatedelete.go (hunk 1 above)
Comment (ketang) [reply cli/templatedelete.go:79]: I think dangerous actions should default to safety.  All yes or all no isn't the appropriate framework; safe versus dangerous is.

Comment (kylecarbs) [reply cli/templatedelete.go:79]: I think it's fine to default to safety here, I'm just promoting consistency among our destructive actions.

Comment (ketang) [reply cli/templatedelete.go:79]: I agree, but the way I want to manifest that agreement is changing the defaults on the others 🙃.  Given the existence of `--yes`, I think it shouldn't get in the way of the #yolo customers.

Comment (ketang) [reply cli/templatedelete.go:79]: #2878

File: cli/templatedelete.go (hunk 2 above)
Comment (ketang) [reply cli/templatedelete.go:77]: done

**Review by kylecarbs — APPROVED** [review]
The PR title should be updated to prefix "fix:" or "feat:". I'd argue this is more of a "fix:" since this resolves a previously broken UX.

LGTM!

**Pull request merged**